samtranslator/model/sam_resources.py

```diff
@@ -867,8 +868,123 @@ def to_cloudformation(self, **kwargs):
             resources.append(role)
             synthetics_canary.ExecutionRoleArn = role.get_runtime_attr("arn")
 
+        if self.CloudWatchAlarms:
+            self._validate_cloudwatch_alarms()
+            for dict_item in self.CloudWatchAlarms:
+                resources.append(self._construct_cloudwatch_alarms(dict_item))
+
         return resources
 
+    def _validate_cloudwatch_alarms(self):
+        """Validates the CloudWatchAlarms property in Serverless Canary
+
+        The property should follow the following structure
+
+        CloudWatchAlarms:
+            - AlarmName:
+                MetricName (required): one of ["SuccessPercent", "Failed", "Duration"]
+                Threshold (optional): any value of type double
+                ComparisonOperator (optional): any of the valid values (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cw-alarm.html#cfn-cloudwatch-alarms-comparisonoperator)
+                Period (optional): Integer that is 10, 30, 60, or any multiple of 60
+            - AlarmName2:
+                MetricName: Failed
+
+        Note: Alarm names are used as logical ids for their respective CloudWatchAlarm property so there should be no
+        duplicate names as we don't want the alarms to override each other without the user knowing
+
+        :raise: InvalidResourceException
+        """
+        # keeps list of alarm names to make sure there are no duplicates
+        list_of_alarm_names = []
+        for dict_item in self.CloudWatchAlarms:
+
+            # Throw an error if there is more than one alarm in the array index, like for example
+            # CloudWatchAlarms:
+            #     - Alarm1:
+            #         MetricName: SuccessPercent
+            #       Alarm2:
+            #         MetricName: SuccessPercent
+            #         Threshold: 90
+            if len(dict_item) > 1:
```

Comment:
```suggestion
            if len(dict_item) != 1:
```

This is because in line 912 we are trying to get the value at index `0`. So I am assuming that this cannot be 0 as well.